jormungandr/src/settings/logging.rs

```diff
@@ -83,125 +75,138 @@ impl FromStr for LogOutput {
+            Option<tracing_appender::non_blocking::WorkerGuard>,
+        ),
+        Error,
+    > {
+        let Self {
+            output,
+            level,
+            format,
+        } = &self;
 
-impl LogOutput {
-    fn to_logger(&self, format: &LogFormat) -> Result<Async, Error> {
-        match self {
-            LogOutput::Stdout => Ok(format.decorate_stdout()),
-            LogOutput::Stderr => Ok(format.decorate_stderr()),
-            #[cfg(unix)]
-            LogOutput::Syslog => {
-                format.require_plain()?;
-                match slog_syslog::unix_3164(Facility::LOG_USER) {
-                    Ok(drain) => Ok(drain.into_async()),
-                    Err(e) => Err(Error::SyslogAccessFailed(e)),
+        let builder = tracing_subscriber::fmt::SubscriberBuilder::default();
+
+        match output {
+            LogOutput::Stdout => {
+                let (subscriber, guard) = tracing_appender::non_blocking(std::io::stdout());
+                let builder = builder.with_writer(subscriber).with_max_level(*level);
+                match format {
+                    LogFormat::Plain => Ok((Box::new(builder.finish()), Some(guard))),
+                    LogFormat::Json => Ok((Box::new(builder.json().finish()), Some(guard))),
                 }
             }
-            #[cfg(unix)]
-            LogOutput::SyslogUdp { host, hostname } => {
-                use std::net::{IpAddr, Ipv4Addr, SocketAddr};
-
-                format.require_plain()?;
-
-                let mut local_port = 30_000;
-                let host = host.parse().map_err(Error::SyslogInvalidHost)?;
-
-                // automatically select local port
-                loop {
-                    let local =
-                        SocketAddr::new(IpAddr::V4(Ipv4Addr::new(127, 0, 0, 1)), local_port);
-                    let res = slog_syslog::SyslogBuilder::new()
-                        .facility(Facility::LOG_USER)
-                        .udp(local, host, hostname)
-                        .start();
-                    match res {
-                        Ok(drain) => break Ok(drain.into_async()),
-                        Err(e) => {
-                            if e.kind() == std::io::ErrorKind::AddrInUse && local_port < 65_535 {
-                                local_port += 1;
-                                continue;
-                            }
-
-                            break Err(Error::SyslogAccessFailed(e));
-                        }
-                    }
+            LogOutput::Stderr => {
+                let (subscriber, guard) = tracing_appender::non_blocking(std::io::stderr());
+                let builder = builder.with_writer(subscriber).with_max_level(*level);
+                match format {
+                    LogFormat::Plain => Ok((Box::new(builder.finish()), Some(guard))),
+                    LogFormat::Json => Ok((Box::new(builder.json().finish()), Some(guard))),
```

Comment:
This logic could be unified.

Reply:
What do you mean here?

Reply:
Handling of stdout and stderr is pretty much the same code, so maybe the match arm should be `LogOutput::Stdout | LogOutput::Stderr` and then a common body. But it can stay as it is, not much of a concern.